src/loss.jl

```diff
@@ -14,13 +21,11 @@ equivalent to `exp.(logp(x)`.
 logp(x; dims=:) = _logp(x; dims=dims)
 
 function logp(x::A; dims=:) where A <: Union{<:KnetArray, Param{<:KnetArray}}
```

Comment:
I'm not yet familiar with AutoGrad's updates, but `Param` seems the only one to work here. I checked that this method gets called (as it should) when doing the following:
```julia
k = KnetArray(rand(2,2))
grad(x->sum(logp(x,dims=1)))(k)
```
@denizyuret what do you think?

Reply:
Inside of @diff, primitive functions produces Result{T} objects. When working KnetArrays, this logp will be called with Result{<:KnetArray}.  Since Result <: Value and Param<:Value, `where A <: Union{<:KnetArray, Value{<:KnetArray}}` can handle both cases.